[Doc] migration guide joda (#51986)
elastic/elasticsearch: docs/reference/migration/migrate_7_0/java_time.asciidoc

the timezone `GMT0` properly anymore. The reason for this is a bug in the
JDK, which has not been fixed for JDK8. You can read more in the
https://bugs.openjdk.java.net/browse/JDK-8138664[official issue]

> The fix is in JDK version 9 and newer.

Reviewer: ```suggestion
The bug is fixed in JDK9 and later versions.
```

With the switch to java time, support for negative timestamps has been removed.
For dates before 1970, use a date format containing a year.

[float]
==== Migration guide

> For a detailed migration guide, see <<migrate_to_java_time>>.

Reviewer: This should fix the docs-ci test.
```suggestion
For a detailed migration guide, see <<migrate-to-java-time>>.
```

With the switch to java time, support for negative timestamps has been removed.
For dates before 1970, use a date format containing a year.

[float]

> ==== Migration Guide

Reviewer: ```suggestion
==== Migration guide
```

You can find a detailed migration guide here

> * <<migrate_to_java_time>>

Reviewer: ```suggestion
For a detailed migration guide, see <<migrate_to_java_time>>.
```
